File: go-sdk/pkg/models/operations/postfilerequestsidcopy.go
package operations

import (
	"openapi/pkg/models/shared"
	"openapi/pkg/utils"
)

type PostFileRequestsIDCopyPathParams struct {
	FileRequestID string `pathParam:"style=simple,explode=false,name=file_request_id"`
}

type PostFileRequestsIDCopyRequest struct {
	Retries    *utils.RetryConfig
	PathParams PostFileRequestsIDCopyPathParams
	Request    *shared.FileRequestCopyRequest `request:"mediaType=application/json"`
}

type PostFileRequestsIDCopyResponse struct {
	ClientError *shared.ClientError
	ContentType string
	FileRequest *shared.FileRequest
	StatusCode  int64
}

File: sdks/ts/src/sdk/models/operations/postfilerequestsidcopy.ts
import { SpeakeasyMetadata, SpeakeasyBase, RetryConfig } from "../../../internal/utils";
import * as shared from "../shared";



export class PostFileRequestsIdCopyPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=file_request_id" })
  fileRequestId: string;
}


export class PostFileRequestsIdCopyRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  retries?: RetryConfig;

  @SpeakeasyMetadata()
  pathParams: PostFileRequestsIdCopyPathParams;

  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  request?: shared.FileRequestCopyRequest;
}


export class PostFileRequestsIdCopyResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  clientError?: shared.ClientError;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  fileRequest?: shared.FileRequest;

  @SpeakeasyMetadata()
  statusCode: number;
}
